File: mock-caas/src/main/java/com/sequenceiq/caas/util/JsonUtil.java
package com.sequenceiq.caas.util;

import java.util.concurrent.Callable;

import javax.annotation.Nonnull;

import org.springframework.stereotype.Component;

import com.fasterxml.jackson.databind.DeserializationFeature;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.sequenceiq.caas.exception.JsonOperationException;

@Component
public class JsonUtil {

    private static final ObjectMapper MAPPER = new ObjectMapper();

    static {
        MAPPER.disable(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES);
    }

    public String toJsonString(Object o) {
        return jsonOperationWithExceptionChange(() -> MAPPER.writeValueAsString(o));
    }

    public  <T> T toObject(String json, Class<T> type) {
        return jsonOperationWithExceptionChange(() -> MAPPER.readValue(json, type));
    }

    private  <T> T jsonOperationWithExceptionChange(@Nonnull Callable<T> callable) {
        try {
            return callable.call();
        } catch (Exception e) {
            throw new JsonOperationException(e);
        }
    }

}

File: cloud-common/src/main/java/com/sequenceiq/cloudbreak/util/JsonUtil.kt
package com.sequenceiq.cloudbreak.util

import java.io.IOException
import java.io.Reader

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.core.TreeNode
import com.fasterxml.jackson.databind.JsonNode
import com.fasterxml.jackson.databind.ObjectMapper

object JsonUtil {

    private val MAPPER = ObjectMapper()

    @Throws(IOException::class)
    fun <T> readValue(reader: Reader, valueType: Class<T>): T {
        return MAPPER.readValue(reader, valueType)
    }

    @Throws(IOException::class)
    fun <T> readValue(content: String, valueType: Class<T>): T {
        return MAPPER.readValue(content, valueType)
    }

    @Throws(JsonProcessingException::class)
    fun writeValueAsString(`object`: Any): String {
        return MAPPER.writeValueAsString(`object`)
    }

    @Throws(IOException::class)
    fun readTree(content: String): JsonNode {
        return MAPPER.readTree(content)
    }

    fun minify(content: String): String {
        try {
            return readTree(content).toString()
        } catch (e: IOException) {
            return "INVALID_JSON_CONTENT"
        }

    }

    @Throws(JsonProcessingException::class)
    fun <T> treeToValue(n: TreeNode, valueType: Class<T>): T {
        return MAPPER.treeToValue(n, valueType)
    }

}
